src/primitives/rectangles.jl

```diff
@@ -14,6 +14,21 @@ end
 ##
 # Constructors & typealiases
 
+"""
+    const Rect{N,T} = HyperRectangle{N,T}
+
+A rectangle in N dimensions, formally the caartesian product of intervals. See also [`HyperRectangle`](@ref). Its aliases are
+
+|Dimensions|Generic    |Float32   |Int       |
+|----------|-----------|----------|----------|
+|Any       |`Rect{N,T}`|`Rectf{N}`|`Recti{N}`|
+|2         |`Rect2{T}` |`Rect2f`  |`Rect2i`  |
+|3         |`Rect3{T}` |`Rect3f`  |`Rect3i`  |
+
+and `RectT{T} = Rect{N,T} where {N}`.
```

Comment:
```suggestion
|Any       |`Rect`  |`Rectf`   |`Recti` |
|2         |`Rect2` |`Rect2f`  |`Rect2i`|
|3         |`Rect3` |`Rect3f`  |`Rect3i`|

For example `Rect2 == (Rect{2,T} where T)` and `Rectf == (Rect{N,Float32} where N)`.
```

In the table I think it's better to show the aliases themselves, which can be used directly by the user and are less confusing to beginners, rather than the generic parametrized types that can only be used with a `where` clause.

Reply:
I think you misunderstand what these aliases are, which means this doc is not well written... I'll rephrase them.

Reply:
What do you mean? You can check in the REPL that what I wrote is correct: `Rect2{T}` by itself is not valid Julia, but `Rect2 == (Rect{2,T} where T)` is exactly right (you can evaluate it, it returns `true`).

Reply:
Why is `Rect2{T}` not valid?
```julia
julia> Rect2{Float64}
Rect2{Float64} (alias for HyperRectangle{2, Float64})
```
While it's true, is there a Julia reference that `Type{A,B} == Type{A}{B}`? I wasn't able to find it and it apparently doesn't work on `Tuple`.

anyway, what I meant was that `RectT` is a standalone alias while you deleted it. I didn't realise that `RectT{T,N}` is basically `Rect{N,T}` and I've updated the docstring.

As for other changes, to me, it's more clear to phrase `Rect2{T} = Rect{2,T}` rather than `Rect2 = Rect{2,T} where T` as the former gives an example on how to use the alias. I'd like to hear opinions from others so I'll leave this unresolved.

Reply:
Sorry I meant that `Rect2{T}` with literal `T` is not valid in itself, i.e. `T` has no special meaning in Julia, you have to define it first (or declare it as a type parameter using `where`). As you say `Rect2{Float64}` is perfectly valid, I think that would be a good example to use here rather than something with a generic `T`, because concrete examples are easier to grasp for beginners. I've edited my suggestion to add it.

Yes the `Type{A,B} == Type{A}{B}` rule is documented here: https://docs.julialang.org/en/v1/manual/types/#UnionAll-Types

> The type application syntax `A{B,C}` requires `A` to be a `UnionAll` type, and first substitutes `B` for the outermost type variable in `A` . The result is expected to be another `UnionAll` type, into which `C` is then substituted. So `A{B,C}` is equivalent to `A{B}{C}` .

I think tuples are a special case because `Tuple{Int}` is already a concrete type so the result doesn't expect an additional type parameter.

> As for other changes, to me, it's more clear to phrase Rect2{T} = Rect{2,T} rather than Rect2 = Rect{2,T} where T as the former gives an example on how to use the alias.

Note that `Rect2` can be used alone, for example `endcorner(r::Rect2) = origin(r) + widths(r)` is valid. But I see your point, I agree your version is better.

Reply:
Then I'll close this conversation.

Off topic: turns out that `Tuple` is not `UnionAll`.
```julia
julia> Tuple <: UnionAll
false
```

Reply:
(please mark this as resolved, I don't see how to do it myself)